precompiles/utils/src/data.rs

```diff
@@ -175,6 +179,14 @@ impl EvmDataWriter {
 		self
 	}
 
+	//TODO is this a good idea?
+	// Write a solidity-style four-byte function selector for the given signature
+	// pub fn write_selector(&mut self, signature: &str) {
+	// 	let selector = &Keccak256::digest(signature)[0..4];
+	// 	I was going to call write_raw_bytes here as a helper.
+	// 	But why do some of these methods take ownership and others not?
```

Comment:
Methods take ownershop so we can call them in a chain `.call(...).call(...)`. I couldn't find back the way to support both styles (taking an `&mut` and return it didn't worked :c)

Reply:
Okay, so I made it compile in . By changing the signature like so. As the designer of the utils, do you think it's a good idea?
```
// Write a solidity-style four-byte function selector for the given signature
pub fn write_selector(mut self, signature: &[u8]) -> Self {
	let selector = &Keccak256::digest(signature)[0..4];
	self.data.extend_from_slice(selector);
	self
}
```